parsec/core/mountpoint/fuse_runner.py

```diff
@@ -167,44 +172,56 @@ def _run_fuse_thread():
             task_status.started(mountpoint_path)
 
     finally:
-        await _stop_fuse_thread(mountpoint_path, fuse_operations, fuse_thread_stopped)
-        event_bus.send(CoreEvent.MOUNTPOINT_STOPPED, **event_kwargs)
-        await _teardown_mountpoint(mountpoint_path)
+        with trio.CancelScope(shield=True) as teardown_cancel_scope:
+            await _stop_fuse_thread(
+                mountpoint_path, fuse_operations, fuse_thread_started, fuse_thread_stopped
+            )
+            event_bus.send(CoreEvent.MOUNTPOINT_STOPPED, **event_kwargs)
+            await _teardown_mountpoint(mountpoint_path)
 
 
 async def _wait_for_fuse_ready(mountpoint_path, fuse_thread_started, initial_st_dev):
     trio_mountpoint_path = trio.Path(mountpoint_path)
-
     # Polling until fuse is ready
     while True:
+        # Tick for 10 ms
         await trio.sleep(0.01)
+        # Stat the root of the workspace
         try:
             stat = await trio_mountpoint_path.stat()
+        # The directory does not exist for some reason
         except FileNotFoundError:
-            continue
+            raise
```

Comment:
What is the point here to catch and raise the exception, explicitness ?

Reply:
Otherwise it's caught by `except OSError: break` which doesn't really make sense if the directory does not exist